### PR TITLE
pkp/pkp-lib#4936 Log ADODB debug messages (rather than displaying)

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

**Comment by asmecher** [comment]
@defstat, would you mind reviewing this? It changes the `debug` setting in `config.inc.php` so that messages are sent to the PHP log rather than the browser. (See the issue description at https://github.com/pkp/pkp-lib/issues/4936.)

File: classes/db/DBConnection.inc.php [hunk 1]
```diff
@@ -56,6 +57,16 @@ function __construct() {
...
+	/**
+	 * Log message handler for ADODB.
+	 * @param $msg string
+	 * @param $newline boolean
+	 * @see ADOConnection::outp
+	 */
+	static function logAdodbMessage($msg, $newline=true) {
+		error_log(Registry::get('system.debug.startTime') . ': ' . PKPString::html2text($msg));
+	}
+
```
Comment (defstat) [inline classes/db/DBConnection.inc.php]: @asmecher It works perfectly fine. Could I suggest 2 things:

- Perhaps adding some prefix for those logs, in order to distinguish from other PHP warnings/errors - Something like: `error_log('PKP-Database-Logger ' . Registry::get('system.debug.startTime') . ': ' . PKPString::html2text($msg));` 
- Perhaps we could add some data regarding the caller of the sql. Something like `
$backtrace = debug_backtrace(2);
error_log(print_r($backtrace, true) . '-> ' . Registry::get('system.debug.startTime') . ': ' . PKPString::html2text($msg));` (even though that could result to a rather extensive log. Maybe we could traverse on the `$backtrace` in order to display less, and more useful info)

Comment (asmecher) [reply classes/db/DBConnection.inc.php]: Thanks, @defstat! I agree with the first; let's come back to the second if there's trouble locating the SQL. I find grepping for a piece of the SQL works pretty well.

**Comment by asmecher** [comment]
Merged to `stable-3_1_2` and `master`!

**Pull request merged**